The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11223875705/job/31199482930) [(plain)](https://github.com/rust-lang/rust/commit/84c5244b0064b243d7e71111f17e92bed79cb68e/checks/31199482930/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.870 Building wheels for collected packages: reuse
#13 2.871   Building wheel for reuse (pyproject.toml): started
#13 3.130   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.131   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.132   Stored in directory: /tmp/pip-ephem-wheel-cache-2_qtdxpf/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.134 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.544 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.544 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.130 Collecting virtualenv
#13 4.130 Collecting virtualenv
#13 4.185   Downloading virtualenv-20.26.6-py3-none-any.whl (6.0 MB)
#13 4.403      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 28.0 MB/s eta 0:00:00
#13 4.450 Collecting distlib<1,>=0.3.7
#13 4.459   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 4.471      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 49.5 MB/s eta 0:00:00
#13 4.514 Collecting filelock<4,>=3.12.2
#13 4.523   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.563 Collecting platformdirs<5,>=3.9.1
#13 4.573   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.655 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.847 Successfully installed distlib-0.3.8 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.26.6
#13 DONE 4.9s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      198592 kB
DirectMap2M:     9238528 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 31.59s
##[endgroup]
fmt check
Diff in /checkout/library/core/tests/nonzero.rs:358:
 #[test]
 fn test_nonzero_fmt() {
     let i = format!("{0}, {0:?}, {0:x}, {0:X}, {0:#x}, {0:#X}, {0:o}, {0:b}, {0:e}, {0:E}", 42);
-    let nz = format!("{0}, {0:?}, {0:x}, {0:X}, {0:#x}, {0:#X}, {0:o}, {0:b}, {0:e}, {0:E}", NonZero::new(42).unwrap());
+    let nz = format!(
+        "{0}, {0:?}, {0:x}, {0:X}, {0:#x}, {0:#X}, {0:o}, {0:b}, {0:e}, {0:E}",
+        NonZero::new(42).unwrap()
 
 
     assert_eq!(i, nz);
Diff in /checkout/library/core/tests/nonzero.rs:365:
+
+
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/tests/iter/adapters/skip.rs" "/checkout/library/core/tests/iter/adapters/cloned.rs" "/checkout/library/core/tests/iter/adapters/step_by.rs" "/checkout/compiler/rustc_feature/src/builtin_attrs.rs" "/checkout/compiler/rustc_feature/src/tests.rs" "/checkout/compiler/rustc_feature/src/unstable.rs" "/checkout/compiler/rustc_feature/src/accepted.rs" "/checkout/compiler/rustc_feature/src/removed.rs" "/checkout/compiler/rustc_feature/src/lib.rs" "/checkout/library/core/tests/iter/traits/iterator.rs" "/checkout/library/core/tests/iter/traits/mod.rs" "/checkout/library/core/tests/iter/traits/accum.rs" "/checkout/library/core/tests/iter/traits/step.rs" "/checkout/library/core/tests/iter/traits/double_ended.rs" "/checkout/library/core/tests/iter/range.rs" "/checkout/library/core/tests/time.rs" "/checkout/library/core/tests/option.rs" "/checkout/compiler/rustc_privacy/src/errors.rs" "/checkout/compiler/rustc_privacy/src/lib.rs" "/checkout/library/core/tests/net/mod.rs" "/checkout/library/core/tests/net/parser.rs" "/checkout/library/core/tests/net/ip_addr.rs" "/checkout/library/core/tests/net/socket_addr.rs" "/checkout/library/core/tests/nonzero.rs" "/checkout/library/core/tests/str_lossy.rs" "/checkout/library/core/tests/fmt/builders.rs" "/checkout/library/core/tests/fmt/mod.rs" "/checkout/library/core/tests/fmt/float.rs" "/checkout/library/core/tests/fmt/num.rs" "/checkout/library/core/tests/hash/sip.rs" "/checkout/library/core/tests/hash/mod.rs" "/checkout/library/core/tests/any.rs" "/checkout/library/core/tests/ascii_char.rs" "/checkout/library/core/tests/str.rs" "/checkout/compiler/rustc_const_eval/src/interpret/intrinsics.rs" "/checkout/compiler/rustc_const_eval/src/interpret/visitor.rs" "/checkout/compiler/rustc_const_eval/src/interpret/mod.rs" "/checkout/compiler/rustc_const_eval/src/interpret/call.rs" "/checkout/compiler/rustc_const_eval/src/interpret/projection.rs" "/checkout/compiler/rustc_const_eval/src/interpret/memory.rs" "/checkout/compiler/rustc_const_eval/src/interpret/util.rs" "/checkout/library/core/tests/num/uint_macros.rs" "/checkout/compiler/rustc_const_eval/src/interpret/cast.rs" "/checkout/compiler/rustc_const_eval/src/interpret/traits.rs" "/checkout/compiler/rustc_const_eval/src/interpret/validity.rs" "/checkout/compiler/rustc_const_eval/src/interpret/step.rs" "/checkout/compiler/rustc_const_eval/src/interpret/intern.rs" "/checkout/compiler/rustc_const_eval/src/interpret/discriminant.rs" "/checkout/compiler/rustc_const_eval/src/interpret/operand.rs" "/checkout/compiler/rustc_const_eval/src/interpret/place.rs" "/checkout/compiler/rustc_const_eval/src/interpret/machine.rs" "/checkout/compiler/rustc_const_eval/src/interpret/eval_context.rs" "/checkout/compiler/rustc_const_eval/src/interpret/operator.rs" "/checkout/compiler/rustc_const_eval/src/interpret/stack.rs" "/checkout/library/core/tests/num/dec2flt/mod.rs" "/checkout/library/core/tests/num/dec2flt/float.rs" "/checkout/library/core/tests/num/dec2flt/parse.rs" "/checkout/library/core/tests/num/dec2flt/lemire.rs" "/checkout/library/core/tests/num/u128.rs" "/checkout/library/core/tests/num/nan.rs" "/checkout/library/core/tests/num/const_from.rs" "/checkout/library/core/tests/num/mod.rs" "/checkout/library/core/tests/num/i16.rs" "/checkout/library/core/tests/iter/adapters/enumerate.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Mon Oct  7 21:07:30 UTC 2024
  network time: Mon, 07 Oct 2024 21:07:30 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/num/nonzero.rs
+++ b/library/core/src/num/nonzero.rs
@@ -110,26 +110,40 @@ impl_zeroable_primitive!(
 pub struct NonZero<T: ZeroablePrimitive>(T::NonZeroInner);
 
 macro_rules! impl_nonzero_fmt {
-    ($Trait:ident) => {
-        #[stable(feature = "nonzero", since = "1.28.0")]
-        impl<T> fmt::$Trait for NonZero<T>
-        where
-            T: ZeroablePrimitive + fmt::$Trait,
-        {
-            #[inline]
-            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                self.get().fmt(f)
+    ($(#[$Attribute:meta] $Trait:ident)*) => {
+        $(
+            #[$Attribute]
+            impl<T> fmt::$Trait for NonZero<T>
+            where
+                T: ZeroablePrimitive + fmt::$Trait,
+            {
+                #[inline]
+                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                    self.get().fmt(f)
+                }
             }
-        }
+        )*
     };
 }
 
-impl_nonzero_fmt!(Debug);
-impl_nonzero_fmt!(Display);
-impl_nonzero_fmt!(Binary);
-impl_nonzero_fmt!(Octal);
-impl_nonzero_fmt!(LowerHex);
-impl_nonzero_fmt!(UpperHex);
+impl_nonzero_fmt! {
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    Debug
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    Display
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    Binary
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    Octal
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    LowerHex
+    #[stable(feature = "nonzero", since = "1.28.0")]
+    UpperHex
+    #[stable(feature = "nonzero_fmt_exp", since = "CURRENT_RUSTC_VERSION")]
+    LowerExp
+    #[stable(feature = "nonzero_fmt_exp", since = "CURRENT_RUSTC_VERSION")]
+    UpperExp
+}
 
 macro_rules! impl_nonzero_auto_trait {
     (unsafe $Trait:ident) => {

--- a/library/core/tests/nonzero.rs
+++ b/library/core/tests/nonzero.rs
@@ -354,3 +354,14 @@ fn test_signed_nonzero_neg() {
     assert_eq!((-NonZero::<i128>::new(1).unwrap()).get(), -1);
     assert_eq!((-NonZero::<i128>::new(-1).unwrap()).get(), 1);
 }
+
+#[test]
+fn test_nonzero_fmt() {
+    let i = format!("{0}, {0:?}, {0:x}, {0:X}, {0:#x}, {0:#X}, {0:o}, {0:b}, {0:e}, {0:E}", 42);
+    let nz = format!(
+        "{0}, {0:?}, {0:x}, {0:X}, {0:#x}, {0:#X}, {0:o}, {0:b}, {0:e}, {0:E}",
+        NonZero::new(42).unwrap()
+    );
+
+    assert_eq!(i, nz);
+}

